src/ariel-os-nrf/src/uart.rs

```diff
@@ -0,0 +1,375 @@
+        }
+    }
+}
+
+fn from_baudrate(baud: Baudrate) -> embassy_nrf::buffered_uarte::Baudrate {
+    match baud {
+        Baudrate::_1200 => embassy_nrf::uarte::Baudrate::BAUD1200,
+        Baudrate::_2400 => embassy_nrf::uarte::Baudrate::BAUD2400,
+        Baudrate::_4800 => embassy_nrf::uarte::Baudrate::BAUD4800,
+        Baudrate::_9600 => embassy_nrf::uarte::Baudrate::BAUD9600,
+        Baudrate::_14_400 => embassy_nrf::uarte::Baudrate::BAUD14400,
+        Baudrate::_19_200 => embassy_nrf::uarte::Baudrate::BAUD19200,
+        Baudrate::_28_800 => embassy_nrf::uarte::Baudrate::BAUD28800,
+        Baudrate::_31_250 => embassy_nrf::uarte::Baudrate::BAUD31250,
+        Baudrate::_38_400 => embassy_nrf::uarte::Baudrate::BAUD38400,
+        Baudrate::_56_000 => embassy_nrf::uarte::Baudrate::BAUD56000,
+        Baudrate::_57_600 => embassy_nrf::uarte::Baudrate::BAUD57600,
+        Baudrate::_76_800 => embassy_nrf::uarte::Baudrate::BAUD76800,
+        Baudrate::_115_200 => embassy_nrf::uarte::Baudrate::BAUD115200,
+        Baudrate::_230_400 => embassy_nrf::uarte::Baudrate::BAUD230400,
+        Baudrate::_250_000 => embassy_nrf::uarte::Baudrate::BAUD250000,
+        Baudrate::_460_800 => embassy_nrf::uarte::Baudrate::BAUD460800,
+        Baudrate::_921_600 => embassy_nrf::uarte::Baudrate::BAUD921600,
+        Baudrate::_1_000_000 => embassy_nrf::uarte::Baudrate::BAUD1M,
+    }
+}
+
+impl From<ariel_os_embassy_common::uart::Baudrate<Self>> for Baudrate {
+    fn from(baud: ariel_os_embassy_common::uart::Baudrate<Self>) -> Baudrate {
+        match baud {
+            ariel_os_embassy_common::uart::Baudrate::Hal(baud) => baud,
+            ariel_os_embassy_common::uart::Baudrate::_2400 => Baudrate::_2400,
+            ariel_os_embassy_common::uart::Baudrate::_4800 => Baudrate::_4800,
+            ariel_os_embassy_common::uart::Baudrate::_9600 => Baudrate::_9600,
+            ariel_os_embassy_common::uart::Baudrate::_19200 => Baudrate::_19_200,
+            ariel_os_embassy_common::uart::Baudrate::_38400 => Baudrate::_38_400,
+            ariel_os_embassy_common::uart::Baudrate::_57600 => Baudrate::_57_600,
+            ariel_os_embassy_common::uart::Baudrate::_115200 => Baudrate::_115_200,
+        }
+    }
+}
+
+/// UART number of data bits.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub enum DataBits {
+    /// 8 bits per character.
+    Data8,
+}
+
+impl From<ariel_os_embassy_common::uart::DataBits<Self>> for DataBits {
+    fn from(databits: ariel_os_embassy_common::uart::DataBits<Self>) -> DataBits {
+        match databits {
+            ariel_os_embassy_common::uart::DataBits::Hal(bits) => bits,
+            ariel_os_embassy_common::uart::DataBits::Data8 => DataBits::Data8,
+        }
+    }
+}
+/// UART number of stop bits.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub enum StopBits {
```

Comment:
We need to mark this as `non_exhaustive`, otherwise adding supporting for the nRF5340 capabilities would be a breaking change:
```suggestion
#[non_exhaustive]
pub enum StopBits {
```

Reply:
Agreed, added in c52f889a